### PR TITLE
chore: migrate type tests to TSTyche

### PR DESCRIPTION
**Pull request by mrazauskas**
Similar to #894

This PR migrates type tests to TSTyche 2.

### CODE REVIEW

File: test/tests/types/basicTypegoose.test-d.ts [hunk 1]
```diff
@@ -139,62 +139,62 @@ async function typeguards() {
...
-        expectType<typegoose.DocumentType<TypeguardsClass>>(someNewDoc.refObjectId);
+        expect(someNewDoc.refObjectId).type.toBe<typegoose.DocumentType<TypeguardsClass>>();
       } else {
-        expectType<typegoose.mongoose.Types.ObjectId | undefined>(someNewDoc.refObjectId);
+        expect(someNewDoc.refObjectId).type.toBe<typegoose.mongoose.Types.ObjectId | undefined>();
       }
 
       if (typegoose.isDocument(someNewDoc.refString)) {
-        expectAssignable<typegoose.DocumentType<TypeguardsClass>>(someNewDoc.refString);
+        expect(someNewDoc.refString).type.toBeAssignableTo<typegoose.DocumentType<TypeguardsClass>>();
```
Comment (mrazauskas) [inline test/tests/types/basicTypegoose.test-d.ts:148]: `.toBeAssignableTo()` is a new matcher that shipped with TSTyche 2. Together with `.toBeAssignableWith()` they make is possible to check assignability both ways. Hence the source value can always stay on the left side of an assertion.

File: tstyche.config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "$schema": "https://tstyche.org/schemas/config.json",
+  "testFileMatch": [
+    "**/*.test-d.ts"
```
Comment (hasezoey) [inline tstyche.config.json:4]: this could be limited to the `test` directory

**Review by hasezoey — CHANGES_REQUESTED** [review]
This looks good to me, i dont think typegoose actually needs to migrate this now, but before it gets conflicts or upgrading typescript later causes problems, i guess it will be better to merge it now.

There are some things i would like to see addressed before merging:
- please re-save all touched `.json` with a ending newline as it was previously
- consider limiting test file matching to the `test` directory

File: tstyche.config.json (hunk 2 above)
Comment (mrazauskas) [reply tstyche.config.json]: Right. This should slightly improve the performance.

**Comment by mrazauskas** [comment]
> i dont think typegoose actually needs to migrate this now

That’s right. There is no dramatic change indeed. The biggest improvement I see: there is no need to have `@tsd/typescript` around. Also no need to keep version of `typescript` and `@tsd/typescript` in sync. Somewhat less hustle with dependencies.

**Comment by mrazauskas** [comment]
Thanks for quick review! Please, ping me if there would be any questions in the future.

Just to remind, TSTyche can run tests using specified version of TypeScript: `tstyche --target 5.4`. This might help to debug types.

It is also possible to use the [`TSTYCHE_TYPESCRIPT_PATH`](https://tstyche.org/reference/environment#typescript-path) environment variable together with `bisect` from [`every-ts`](https://github.com/jakebailey/every-ts) to find out which commit to TypeScript repo broke something.

**Pull request merged**